internal/controller/common/in_progress_strategy.go

```diff
@@ -119,6 +121,32 @@ func (store *inProgressStrategyStore) SetStrategy(namespacedName k8stypes.Namesp
 	store.mutex.Unlock()
 }
 
+// ExtractOriginalScaleMinMax returns a JSON string of the scale definition
+// including only min and max fields extracted from the given unstructured object.
+func ExtractOriginalScaleMinMax(object map[string]any, pathToScale []string) (*string, error) {
+	originalScaleDef, foundScale, err := unstructured.NestedMap(object, pathToScale...)
+	if err != nil {
+		return nil, err
+	}
+
+	if !foundScale {
+		return nil, nil
+	}
+
+	originalScaleMinMaxOnly := map[string]any{
+		"min": originalScaleDef["min"],
+		"max": originalScaleDef["max"],
+	}
```

Comment:
so, basically if "min" and "max" aren't set at all, then it's something like `{"min": 0, "max": 0 }`?